### ISSUE TITLE
support Phanpy embeds

### ISSUE THREAD

**Issue by wakest** [stefanbohacek/fediverse-embeds-wordpress-plugin#29]
when using the fediverse client Phanpy, they have their own embed code so when pasting from there you could style it potentially the same as when copied from the Mastodon web front end.

or maybe it makes more sense to get Phanpy to show an option to copy the normal Mastodon embed code

<img width="759" alt="image" src="https://github.com/user-attachments/assets/0785df50-93bd-46f9-82e6-1a828afd3052">

either way its confusing rn cause I use Phanpy as my client and went to copy the embed code from a post and it took me a while to figure out why it wasn't working with your plugin

**Comment by stefanbohacek** [comment]
@wakest That is very interesting, I'll look into it!

Thank you for opening an issue for this!

**Comment by stefanbohacek** [comment]
@cheeaun Would you be interested in adding a class to the `blockquote` element that would help identify a Phanpy embed?

**Comment by cheeaun** [comment]
@stefanbohacek wondering, what would be a good class name?

**Comment by stefanbohacek** [comment]
@cheeaun One of the [two hard problems](https://skeptics.stackexchange.com/questions/19836/has-phil-karlton-ever-said-there-are-only-two-hard-things-in-computer-science) in computer science!

Mastodon uses `mastodon-embed`, so perhaps `phanpy-embed` will do?

**Comment by cheeaun** [comment]
@stefanbohacek `phanpy-embed` sounds kinda weird as it's bound to a client, imagine there's `ivory-embed` or `ice-cubes-embed` in the future 😂

Thinking if a generic, future-proof, classless option like `data-fediverse-embed-url=[URL]` would make more sense?

**Comment by stefanbohacek** [comment]
Fair point! Maybe you could do both the class and a data attribute? I think it makes sense to follow Mastodon's example and identify the source of the embed, plus also use something more generic.

**Comment by cheeaun** [comment]
Hmm, I try not to include classes as there's potential for clash. `[data-]*` is still the safest, I think.

Phanpy is also not Mastodon-only, as it can work with Pixelfed, *key, Hometown, GTS, etc, which means it doesn't seem right to use `mastodon-embed`. I'm also not sure how you're fetching the post for the non-Mastodon instances.

But I guess besides `data-fediverse-embed-url`, this plugin also need something like `data-fediverse-server="mastodon"` right? So that you can check which server software and apply specific logic for it?

**Comment by stefanbohacek** [comment]
Sorry, to clarify, by following Mastodon's example I meant you could include `phanpy-embed` to identify this element as an embed code retrieved via Phanpy, not to use the actual `mastodon-embed` class.

Either way, I do like your idea of including the name of the platform via additional data attribute, this might even be a good feature to request for Mastodon (and other platforms, when they support embeds)!

**Comment by cheeaun** [comment]
I gave this some thought and seems like `data-fediverse-server="mastodon"` is not quite feasible, at least from client-side, as it needs to query the domain (webfinger or node-info) to know what server software it's running on - which can be (or should be?) done from this plugin instead? 🤔

Anyway, I could add `data-fediverse-embed-url`. Or… even more generic (and shorter) `data-fediverse-url`?

**Comment by cheeaun** [comment]
FYI `data-source="fediverse"` change is now deployed on Phanpy production.

Looks like this:
![Screenshot 2024-10-11 at 8 45 06 PM](https://github.com/user-attachments/assets/5d675021-5e61-4ef0-97e9-00bf6122be74)

**Comment by stefanbohacek** [comment]
@cheeaun Looks good!

I'll work on adding the support over the next few days. Thank you!

**Comment by wakest** [comment]
did this ever get figured out @stefanbohacek?
